etor/src/main/java/gov/hhs/cdc/trustedintermediary/external/database/DatabasePartnerMetadataStorage.java

```diff
@@ -35,12 +45,25 @@ public static DatabasePartnerMetadataStorage getInstance() {
     public Optional<PartnerMetadata> readMetadata(final String uniqueId)
             throws PartnerMetadataException {
         try {
-            PartnerMetadata data = (PartnerMetadata) dao.fetchMetadata(uniqueId);
-            return Optional.ofNullable(data);
+            PartnerMetadata metadata =
+                    dao.fetchFirstData(
+                            connection -> {
+                                try {
+                                    PreparedStatement statement =
+                                            connection.prepareStatement(
+                                                    "SELECT * FROM metadata where received_message_id = ? OR sent_message_id = ?");
```

Comment:
At some point I think we should separate SQL code from Java code. In the meantime I've been placing the sql code in a variable at the start of the method where it's used, but we should probably start putting them in a more centralized place soon. It could be external files or java constants